### DISCUSSION TITLE
利用CommunityToolkit.Mvvm自动生成属性,编译无法识别,自动生成的代码,无法识别生成的属性,为什么?

### DISCUSSION THREAD

**Discussion by huangzhezhizhi** [BYJRK/DotNet-Discussions#326] [C# 技术交流]
[ObservableProperty]
private ViewModelBase? currentViewModel;
public MainViewModel(NavigationService navigationService)
{
    navigator = navigationService;
    navigator.CurrentViewModelChanged += () =>
    {
        CurrentViewModel = navigationService.CurrentViewModel;
    };
    // navigate to login page at first
    navigator.NavigateTo<LoginViewModel>();
} 编译报找不到CurrentViewModel的定义,1>D:\Users\xiao'mi\source\repos\Navgateion\Navgateion\ViewModel\MainViewModel.cs(29,17,29,33): error CS0103: 当前上下文中不存在名称“CurrentViewModel”

**Comment by CodingOctocat** [comment]
可能是 VS 17.11.0 的 bug？重启 VS 应该就可以了。

**Comment by huangzhezhizhi** [comment]
有这么简单就好了



---原始邮件---
发件人: ***@***.***&gt;
发送时间: 2024年8月17日(周六) 中午11:18
收件人: ***@***.***&gt;;
抄送: ***@***.******@***.***&gt;;
主题: Re: [BYJRK/DotNet-Discussions] 利用CommunityToolkit.Mvvm自动生成属性,编译无法识别,自动生成的代码,无法识别生成的属性,为什么? (Discussion #326)




 
可能是 VS 17.11.0 的 bug？重启 VS 应该就可以了。

**Reply by CodingOctocat** [reply]
我升级 VS 后，源代码生成器就有点问题了，以前秒生成，现在明显滞后。

**Comment by huangzhezhizhi** [comment]
加了，我是2022版的，很奇怪，生成文件代码都有，就是编译报错



---原始邮件---
发件人: ***@***.***&gt;
发送时间: 2024年8月18日(周日) 晚上10:23
收件人: ***@***.***&gt;;
抄送: ***@***.******@***.***&gt;;
主题: Re: [BYJRK/DotNet-Discussions] 利用CommunityToolkit.Mvvm自动生成属性,编译无法识别,自动生成的代码,无法识别生成的属性,为什么? (Discussion #326)




 
有没有添加 x:DataType="xxxViewModel" ；
 有没有添加
 <Design.DataContext&gt; 	<vm:ViewModel /&gt; </Design.DataContext&gt;

**Comment by huangzhezhizhi** [comment]
我目前是手动的，没有自动



---原始邮件---
发件人: ***@***.***&gt;
发送时间: 2024年8月19日(周一) 上午10:25
收件人: ***@***.***&gt;;
抄送: ***@***.******@***.***&gt;;
主题: Re: [BYJRK/DotNet-Discussions] 利用CommunityToolkit.Mvvm自动生成属性,编译无法识别,自动生成的代码,无法识别生成的属性,为什么? (Discussion #326)




 
我升级 VS 后，源代码生成器就有点问题了，以前秒生成，现在明显滞后。
